resilience4j-cache/src/main/java/io/github/resilience4j/cache/CacheRegistryStore.java

```diff
@@ -0,0 +1,79 @@
+package io.github.resilience4j.cache;
+
+import io.github.resilience4j.core.RegistryStore;
+
+import javax.cache.Cache;
+import javax.cache.processor.EntryProcessor;
+import javax.cache.processor.EntryProcessorException;
+import javax.cache.processor.MutableEntry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class CacheRegistryStore<E> implements RegistryStore<E> {
+
+    private final Cache<String, E> cacheStore;
+
+    public CacheRegistryStore(Cache<String, E> cacheStore) {
+        this.cacheStore = cacheStore;
+    }
+
+    @Override
+    public E computeIfAbsent(String key, Function<? super String, ? extends E> mappingFunction) {
+        try {
+            return cacheStore.invoke(key, new AtomicComputeProcessor<>(), mappingFunction);
+        } catch (EntryProcessorException e) {
+            throw new RuntimeException(e.getCause());
```

Comment:
I think we should throw a custom exception:

Instead of wrapping it in a `RuntimeException` or `CacheException`, we could define a `CacheRegistryStoreException`.

Reply:
Thanks for the review!

I didn't create a new type to keep the behavior in sync with the in-memory store that throws underlying ConcurrentHashMap exception type: https://github.com/resilience4j/resilience4j/blob/3dc2eadb654514b5d503317dbf80a776a892931c/resilience4j-core/src/main/java/io/github/resilience4j/core/registry/InMemoryRegistryStore.java#L35

Do you think it makes sense to create a `RegistryStoreException` that can be used by all store implementations?

Reply:
>      * @throws RuntimeException or Error if the remappingFunction does so,
     *         in which case the mapping is unchanged

Yes, you are right. Then let's keep it as it is.